### PR TITLE
NG: Switch to modern responsive 3-column layout theme `sphinx-basic-ng`

### PR DESCRIPTION
**Pull request by amotl**
## About

The foundation of the current page design is based on Sphinx' original `basic/layout.html`, _Copyright 2007-2014 by the Sphinx team_, so it is already 16 years old, and deserves a modernization.

Following up on GH-378, this patch starts deriving from [sphinx-basic-ng](https://sphinx-basic-ng.readthedocs.io/), in order to bring in more thorough modernizations, also on the page layout level.

## Preview
https://crate-docs-theme--390.org.readthedocs.build/en/390/

## Details

It starts with an empty scaffolded canvas. -- 62f971d2254

image

/cc @surister, @simonprickett

### CODE REVIEW

**Comment by amotl** [comment]
## Progress

### With header and footer

Now also uses CSS directive `position: sticky` instead of `position: fixed`. -- 9fb507e58a532, e135bd44ad1

image

File: src/crate/theme/rtd/crate/static/css/ng/page-tools.sass [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+.version-chooser-link,
+.feedback-compact-link
+  color: #222222
+
+/* Attempt to fix jumping border */
+/*
+ details.sd-dropdown:not([open]).sd-card
+  border: 1px solid transparent
+ */
```
Comment (amotl) [inline src/crate/theme/rtd/crate/static/css/ng/page-tools.sass:37]: This suggestion by @msbt improves the border jumping, but introduces other wobbles.

File: src/crate/theme/rtd/crate/static/vendor/furo/styles/_scaffold.sass [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+  scrollbar-width: thin
+  scrollbar-color: var(--color-foreground-border) transparent
+
+  // Override Chrome scrollbar styles
+  &::-webkit-scrollbar
+    width: 0.25rem
+    height: 0.25rem
+  &::-webkit-scrollbar-thumb
+    background-color: var(--color-foreground-border)
+    border-radius: 0.125rem
```
Comment (proddata) [inline src/crate/theme/rtd/crate/static/vendor/furo/styles/_scaffold.sass:36]: That is leading to the scrollbars showing in Chrome
image


They are not showing up in Safari
image

Comment (amotl) [reply src/crate/theme/rtd/crate/static/vendor/furo/styles/_scaffold.sass:36]: Thank you for investigating and reporting this. @msbt: Can you take care on behalf of your working branch/PR at GH-506?

Comment (amotl) [reply src/crate/theme/rtd/crate/static/vendor/furo/styles/_scaffold.sass:36]: Fix already arrived per cdffe7a8d3f2f9072b7e? That was quick!

Comment (msbt) [reply src/crate/theme/rtd/crate/static/vendor/furo/styles/_scaffold.sass:36]: @amotl @proddata I didn't have the issue in Firefox on Linux, could reproduce it with Chromium thought, is that better now?

Comment (amotl) [reply src/crate/theme/rtd/crate/static/vendor/furo/styles/_scaffold.sass:36]: @proddata confirmed it is better now. Thank you.

**Comment by amotl** [comment]
Hi again. @msbt contributed significantly to this patch. Thank you very much!

- GH-506

We will integrate those improvements here, and run a devX release, in order to be able to slap the new theme on a few downstream projects.

**Comment by amotl** [comment]
Originally submitted on Jul 22, 2023, more than a year ago, this was quite a long-running patch once again, traveling through marvellous journeys in exciting times. Thank you so much for your support to make it converge well.

We hope all relevant/important teething issues have been resolved so far. If you can spot any other rendering flaws or other misbehaviours, or if you want to share suggestions for further improval, please tell us on a ticket here, or on company Slack. Thank you.

**Pull request merged**